File: solutions/RSync/src/main/scala/sync/tt/TruthTable.scala
package sync.tt

class TruthTable(protected var name: String, protected var rows: Set[Row], protected var ports: Set[Port], l_Location: String) extends LocatedElement(l_Location) {

  def getName(): String = {
    name
  }

  def setName(n: String): Unit = {
    require(n != null)
    name = n
    +this syncSetName ()
  }

  def getRows(): Set[Row] = {
    rows
  }

  def addRows(r: Row): Unit = {
    require(r != null)
    require(!rows.contains(r))
    rows += r
    +this syncAddRows (r)
  }

  def removeRows(r: Row): Unit = {
    require(r != null)
    require(rows.contains(r))
    rows -= r
    +this syncRemoveRows (r)
  }

  def getPorts(): Set[Port] = {
    ports
  }

  def addPorts(p: Port): Unit = {
    require(p != null)
    require(!ports.contains(p))
    ports += p
    +this syncAddPorts (p)
  }

  def removePorts(p: Port): Unit = {
    require(p != null)
    require(ports.contains(p))
    ports -= p
    +this syncRemovePorts (p)
  }

  override def toString(): String = {
    "TruthTable:" + " name=" + name + " location=" + location
  }

}



    

File: solutions/EMFSolutionYAMTL/src/main/emf-gen/tt/TruthTable.java
/**
 */
package tt;

import org.eclipse.emf.common.util.EList;

/**
 * <!-- begin-user-doc -->
 * A representation of the model object '<em><b>Truth Table</b></em>'.
 * <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 *   <li>{@link tt.TruthTable#getName <em>Name</em>}</li>
 *   <li>{@link tt.TruthTable#getPorts <em>Ports</em>}</li>
 *   <li>{@link tt.TruthTable#getRows <em>Rows</em>}</li>
 * </ul>
 *
 * @see tt.TTPackage#getTruthTable()
 * @model
 * @generated
 */
public interface TruthTable extends LocatedElement {
	/**
	 * Returns the value of the '<em><b>Name</b></em>' attribute.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Name</em>' attribute isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Name</em>' attribute.
	 * @see #setName(String)
	 * @see tt.TTPackage#getTruthTable_Name()
	 * @model unique="false" required="true" ordered="false"
	 * @generated
	 */
	String getName();

	/**
	 * Sets the value of the '{@link tt.TruthTable#getName <em>Name</em>}' attribute.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Name</em>' attribute.
	 * @see #getName()
	 * @generated
	 */
	void setName(String value);

	/**
	 * Returns the value of the '<em><b>Ports</b></em>' containment reference list.
	 * The list contents are of type {@link tt.Port}.
	 * It is bidirectional and its opposite is '{@link tt.Port#getOwner <em>Owner</em>}'.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Ports</em>' containment reference list isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Ports</em>' containment reference list.
	 * @see tt.TTPackage#getTruthTable_Ports()
	 * @see tt.Port#getOwner
	 * @model opposite="owner" containment="true" required="true" ordered="false"
	 * @generated
	 */
	EList<Port> getPorts();

	/**
	 * Returns the value of the '<em><b>Rows</b></em>' containment reference list.
	 * The list contents are of type {@link tt.Row}.
	 * It is bidirectional and its opposite is '{@link tt.Row#getOwner <em>Owner</em>}'.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Rows</em>' containment reference list isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Rows</em>' containment reference list.
	 * @see tt.TTPackage#getTruthTable_Rows()
	 * @see tt.Row#getOwner
	 * @model opposite="owner" containment="true" lower="2" ordered="false"
	 * @generated
	 */
	EList<Row> getRows();

} // TruthTable
